torch/_dynamo/variables/user_defined.py

```diff
@@ -1495,6 +1478,56 @@ def unpack_var_sequence(self, tx):
             return self._dict_vt.unpack_var_sequence(tx)
         raise NotImplementedError
 
+    def is_modified(self, side_effects):
+        return side_effects.is_attribute_mutation(self) or side_effects.is_modified(
+            self._dict_vt
+        )
```

Comment:
I think `side_effects.is_attribute_mutation(self)` would always return true for `UserDefinedListVariable` etc.?
Doesn't that cause redundant codegen calls if the underlying user-defined list was never modified?

Reply:
I see; in that case the better abstraction to use might be just `side_effects.is_modified(self)`.